src/buildStyles.mjs

```diff
@@ -0,0 +1,87 @@
+ * - not support a central configuration of the output directory (instead the out path must be
+ * specified for every file)
+ * - not watch included files (chould be done through e.g. chokidar)
+ */
+const buildStyles = async({
+    destinationFolder = '.',
+    sourceFolder = '.',
+    sourceFiles = [],
+    compress = false,
+    showNotifications = false,
+} = {}) => {
+
+    // De-glob files
+    const files = sourceFiles
+        .map((file) => join(sourceFolder, file))
+        .map((fileWithPath) => glob.sync(fileWithPath))
+        .flat();
+
+    // Process files
+    const parsedCSSPromises = files.map(async(file) => {
+
+        const cssFileName = basename(file).replace(/\.scss$/, '.css');
+        const destinationFilePath = join(destinationFolder, cssFileName);
+
+        const sassOptions = { sourceMap: true, style: (compress ? 'compressed' : 'expanded') };
+        const sassResult = sass.compile(file, sassOptions);
+
+        // prev: Pass a previous sourceMap object, see docs for SourceMapOptions at
+        // https://postcss.org/api/; comment here
+        // https://github.com/postcss/postcss/issues/222#issuecomment-318136962
+        // If from or to are not provided, warning will be shown because sourceMaps cannot be
+        // generated correctly (as they contain the css file's name)
+        // TBD (get some experience first): We might also handle PostCSS via bash/console
+        const postCSSOptions = {
+            from: file,
+            to: destinationFilePath,
+            map: { prev: sassResult.sourceMap },
+        };
+        const postCSSResult = await postcss([autoprefixer]).process(sassResult.css, postCSSOptions);
+        postCSSResult.warnings().forEach((warning) => console.warn(warning.toString()));
+
+        return {
+            css: postCSSResult.css,
+            sourceMap: sassResult.sourceMap,
+            destinationFilePath,
+        };
+
+    });
+    const parsedCSSInfo = await Promise.all(parsedCSSPromises);
+
+
+    // Write files to file system
+    parsedCSSInfo.forEach((fileInfo) => {
+        // writeFile fails if directory does not exist; therefore create the directory
+        // (recursive:true does only create inexisting folders)
+        mkdirSync(dirname(fileInfo.destinationFilePath), { recursive: true });
+        writeFileSync(fileInfo.destinationFilePath, fileInfo.css);
+        // A CSS map has a .map ending and contains the JSONified content of result.sourceMap
+        // returned by sass.compile()
+        writeFileSync(`${fileInfo.destinationFilePath}.map`, JSON.stringify(fileInfo.sourceMap));
+    });
+
+
+    // Display notification
+    if (showNotifications) {
+        const totalSize = parsedCSSInfo.reduce((sum, fileInfo) => sum + fileInfo.css.length, 0);
+        const totalSizeInKB = Math.ceil(totalSize / 1000);
+        notifier.notify({
+            title: 'Styles Done 💄',
+            message: `Built ${files.length} CSS files (total size: ${totalSizeInKB}KB)`,
```

Comment:
* Kann man die Notifications für JS und Styles noch etwas vereinheitlichen? Oder ist der Unterschied absichtlich?
* Kann man für Styles keine Fehler in der Konsole loggen?

Reply:
1. Notifications: Das ginge – `eslint` generiert die Files aber selber, während wir sie bei SASS manuell schreiben müssen. Deswegen ist's deutlich einfacher, bei SASS die Dateigrösse rasch auszulesen; bei JavaScript fehlen die Informationen teils (kommen insb. nicht daher als Return-Value bei `build`). Ich würd's deswegen mal so belassen – oder wär's hilfreich, wenn ihr bei JS auch die Filesize hättet?
2. Fehler: `.compile` gibt ein `CompileResult` zurück, das gemäss [Doku](https://sass-lang.com/documentation/js-api/interfaces/CompileResult) über keine Fehler verfügt. Ich gehe davon aus, dass SASS direkt loggt. Müssen wir beim Einsatz herausfinden.